src/routes.rs

```diff
@@ -2,20 +2,26 @@ use std::str;
 
 use client_interface::ClientInterface;
 use response::empty::Empty;
+use response::authenticate::Authenticate;
+use response::trowtoken::TrowToken;
 use response::errors::Error;
 use response::html::HTML;
 use response::upload_info::UploadInfo;
 use rocket::request::{self, FromRequest, Request};
 use rocket::{self, Outcome};
 use rocket_contrib::json::Json;
 use TrowConfig;
-
+use serde_json::Value;
+use crypto::sha2::Sha256;
+use jwt::{Header, Token, Registered};
 use types::*;
+const AUTHORISATION_SECRET: &str = "Bob Marley Rastafaria";
```

Comment:
Yeah, we're going to have add the secret stuff as state to Rocket so you can retrieve it from any request

Reply:
Let's do that in a separate PR